lib/bridge.js

```diff
@@ -742,3 +843,36 @@ BridgeContext.prototype.get = function(roomStore, userStore) {
  * @param {string} line The text to be logged.
  * @param {boolean} isError True if this line should be treated as an error msg.
  */
+
+ /**
+  * @typedef Bridge~thirdPartyLookup
+  * @type {Object}
+  * @property {string[]} protocols Optional list of recognised protocol names.
+  * If present, lookups for unrecognised protocols will be automatically
+  * rejected.
+  * @property {Bridge~getProtocol} getProtocol Function. Called for requests
+  * for 3PE query metadata.
+  * @property {Bridge~getLocation} getLocation Function. Called for requests
+  * for 3PLs.
+  * @property {Bridge~getUser} getUser Function. Called for requests for 3PUs.
+  */
+
+ /**
+  * Invoked on requests for 3PE query metadata
+  * @callback Bridge~getProtocol
+  * @param {string} protocol The name of the 3PE protocol to query
+  */
+
+ /**
+  * Invoked on requests for 3PLs
+  * @callback Bridge~getLocation
+  * @param {string} protocol The name of the 3PE protocol
+  * @param {Object} fields The location query field data
+  */
+
+ /**
+  * Invoked on requests for 3PUs
+  * @callback Bridge~getUser
+  * @param {string} protocol The name of the 3PE protocol
+  * @param {Object} fields The user query field data
```

Comment:
The fields are defined by whatever makes sense to each protocol individually. The choice of `protocol` effectively decides what fields are valid - that's the metadata returned by `getProtocol`.